hr_employee_attendance_report/tests/test_hr_employee_attendance_report.py

```diff
@@ -0,0 +1,103 @@
+                "overtime_start_date": datetime(2021, 1, 1),
+                "hr_attendance_overtime": True,
+                "overtime_company_threshold": 10,
+                "overtime_employee_threshold": 10,
+            }
+        )
+        cls.user = new_test_user(
+            cls.env,
+            login="fru",
+            groups="base.group_user,hr_attendance.group_hr_attendance_manager",
+            company_id=cls.company.id,
+        ).with_company(cls.company)
+        cls.employee = cls.env["hr.employee"].create(
+            {
+                "name": "Demo Emp",
+                "user_id": cls.user.id,
+                "company_id": cls.company.id,
+            }
+        )
+
+        cls.leave_type = cls.env["hr.leave.type"].create(
+            {
+                "name": "Time Off",
+                "time_type": "leave",
+                "requires_allocation": "yes",
+                "allocation_validation_type": "no",
+            }
+        )
+        cls.leave = cls.env["hr.leave"].create(
+            {
+                "name": "Leave test",
+                "holiday_status_id": cls.leave_type.id,
+                "date_from": datetime(2021, 1, 2, 1, 0, 0),
+                "date_to": datetime(2021, 1, 5, 23, 0, 0),
+                "request_date_from": datetime(2021, 1, 2, 8, 0, 0),
+                "request_date_to": datetime(2021, 1, 5, 17, 0, 0),
+                "number_of_days": 1,
+                "employee_id": cls.employee.id,
+            }
+        )
+        cls.allocation = cls.env["hr.leave.allocation"].create(
+            {
+                "name": "Some Holiday",
+                "holiday_type": "employee",
+                "employee_ids": [(4, cls.employee.id)],
+                "employee_id": cls.employee.id,
+                "holiday_status_id": cls.leave_type.id,
+                "number_of_days": 2,
+                "allocation_type": "regular",
+                "date_from": datetime(2021, 1, 2, 8),
+                "date_to": datetime(2021, 1, 3, 17),
+            }
+        )
+        cls.resource_leave = cls.env["resource.calendar.leaves"].create(
+            [
+                {
+                    "name": "Absence",
+                    "company_id": cls.company.id,
+                    "calendar_id": cls.employee.resource_calendar_id.id,
+                    "resource_id": cls.employee.resource_id.id,
+                    "holiday_id": cls.leave.id,
+                    "date_from": datetime(2021, 1, 2, 8, 0, 0),
+                    "date_to": datetime(2021, 1, 2, 17, 0, 0),
+                    "time_type": "leave",
+                }
+            ]
+        )
+
+        cls.env["hr.attendance"].create(
+            {
+                "employee_id": cls.employee.id,
+                "check_in": datetime(2021, 1, 4, 8, 0),
+                "check_out": datetime(2021, 1, 4, 20, 0),
+            }
+        )
+        cls.env["hr.attendance.overtime"].create(
```

Comment:
these records, if you don't re-use them, don't add them in `def setUpClass(cls)` but your test function.